packages/taco/src/conditions/base/json-api.ts

```diff
@@ -0,0 +1,28 @@
+import { z } from 'zod';
+
+import { Condition } from '../condition';
+import {
+  OmitConditionType,
+  returnValueTestSchema,
+} from '../shared';
+
+export const JsonApiConditionType = 'json-api';
+
+export const JsonApiConditionSchema = z.object({
+  conditionType: z.literal(JsonApiConditionType).default(JsonApiConditionType),
+  endpoint: z.string().url(),
+  parameters: z.record(z.string(), z.unknown()).optional(),
+  query: z.string().optional(),
```

Comment:
I'm not sure how to implement this, but this field is expected to be a valid JSONPath query.  It may be helpful to consumers to have a taco-web level validation of these queries to prevent invalid conditions from by created.  Perhaps this can be done with a jsonpath library or a rudimentary regex.

Reply:
Json path strings have a specific format/syntax. It would be good to ensure that the format/syntax is correct at construction time. Perhaps https://www.npmjs.com/package/jsonpath / https://www.npmjs.com/package/@types/jsonpath can be of use for validation. Here's an example we did for SIWE messages - https://github.com/nucypher/taco-web/pull/527/files#diff-979bf898c06acdb90eb18e31377b0d4423990d3299e14a7596b3efca73272b49R20.

Reply:
> as i mentioned on the call

Ah, perhaps I missed that comment, my bad.

>  The path only becomes valid/invalid when there is some json associated with it

This is a good point for syntactically valid expressions, however...

> any string is a valid json path

This is not true since it's possible to create syntactically invalid json path expressions.  Nonetheless, this is not a blocking issue but does represent a small improvement in developer experience.  Here are some reasonable (imo) examples:

`$.store.book[?(@.price < ]`
 The filter expression is incomplete and not properly closed.

`$[store][book]`
Incorrect usage of square brackets. Proper syntax should be `$.store.book` or `$.store['book']`.

`$.store.book[*`
Asterisk wildcard is not properly closed with a bracket. It should be `$.store.book[*]`.

Reply:
this is incorrect, without any json to compare with `$.store.book[?(@.price < ]` is a valid json path, because the key in the dictionary could be the string `"book[?(@.price < ]"`

I have gone through this with some of the json path libraries already

`$[store][book]` again, the key could be the string "[store][book]"

Reply:
In order to parse the string literal `book[?(@.price < ]` as a dictionary key that part of the expression must be enclosed in qutation marks?  Something like `$.store['book[?(@.price < ]']`

Reply:
ok, i'll add it back in and some tests

Reply:
yeah i understand this - my point is that if the field `Fields('book[?(@.price < ]')` exists then it's a valid json path

Reply:
maybe when I had this in my testing examples where wrong, but I've added it back in https://github.com/nucypher/taco-web/pull/550/commits/4546b2f6503da046ec275748c79e0e42d1673ff1
Currently it fails, due to the path being valid without any json to compare with

Reply:
> yeah i understand this - my point is that if the field `Fields('book[?(@.price < ]')` exists then it's a valid json path

It's only syntactically valid if the expression has the correct enclosing quotes (`'$.store[\'book[?(@.price < ]\']'`).  Also consider unclosed square brackets for example (`'$.store.book[*'` is syntactically invalid but can be made valid by adding the appropriate quote and square bracket `'$.store[\'book[*\']'` is valid).   

My point is: it is possible to create syntactically invalid jsonpath expressions.  There's nothing we can do to validate that the expression will be valid against a JSON endpoint's response but we can easily detect syntax errors.